File: 100 days of leetcode/Day17.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int[] answer = new int[nums.length];

        //Finding if there are two zeros or more
        int count = 0;
        for(int i = 0; i < nums.length; i++){
            if(nums[i] == 0){
                count++;
            }
        }

        if(count > 1){ // If number of zeros is greater than two
            for(int i = 0; i < nums.length; i++){
                answer[i] = 0;
            }
        }
        else if(count == 1){ // If number of zero is one
            int temp = 1;   
            int index = 0;
            for(int i = 0; i < nums.length; i++){
                if(nums[i] != 0){
                    temp *= nums[i];
                }
                else{
                    index = i;
                }
                answer[i] = 0;
            }
            answer[index] = temp;
            // answer.push_back(0);
        }
        else{ // if there is no zero element
            int temp = 1;
            for(int i = 0; i < nums.length; i++){
                temp *= nums[i];
            }
            for(int i = 0; i < nums.length; i++){
                answer[i] = temp/nums[i];
            }
        }

        return answer;
    }
};

File: 100 days of leetcode/Day17.py
class Solution(object):
    def productExceptSelf(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """

        answer = []

        # Finding if there are two zeros or more
        count = 0
        for i in range(len(nums)):
            if(nums[i] == 0):
                count += 1

        if(count > 1):  # If number of zeros is greater than two
            for i in range(len(nums)):
                answer.append(0)

        elif(count == 1): # If number of zero is one
            temp = 1
            index = 0
            for i in range(len(nums)):
                if(nums[i] != 0):
                    temp *= nums[i]
                else:
                    index = i
                answer.append(0)
            answer[index] = temp
        else: # if there is no zero element
            temp = 1
            for i in range(len(nums)):
                temp *= nums[i]
            for i in range(len(nums)):
                answer.append(temp/nums[i])

        return answer
    